File: gameSrc/model (2b)/tile.js
/**
 *
 * @class
 */
cwt.Tile = my.Class({

  constructor: function () {
    this.unit = null;
    this.property = null;
    this.type = null;
  },

  /**
   *
   * @param unit
   */
  setUnit: function ( unit ) {
    if( DEBUG ) assert( unit instanceof cwt.Unit );

    this.unit = unit;
  },

  /**
   *
   * @param property
   */
  setProperty: function ( property ) {
    if( DEBUG ) assert( property instanceof cwt.Property );

    this.property = property;
  },

  /**
   *
   * @return {Boolean}
   */
  isOccupied: function () {
    return this.unit !== null;
  }

});

File: gameSrc/org/wolftec/cwt/model/Tile.java
package org.wolftec.cwt.model;

import org.wolftec.cwt.sheets.TileType;

public class Tile {

  public TileType type;
  public Unit     unit;
  public Property property;

  public int      variant         = 0;

  public int      visionTurnOwner = 0;
  public int      visionClient    = 0;

  public Tile() {
    this.type = null;
    this.unit = null;
    this.property = null;
    this.visionTurnOwner = 0;
    this.variant = 0;
    this.visionClient = 0;
  }

  public boolean isOccupied() {
    return unit != null;
  }

  public boolean isVisible() {
    return visionTurnOwner > 0;
  }

}
